src/mlpack/methods/ann/loss_functions/binary_cross_entropy_loss_impl.hpp

```diff
@@ -13,42 +13,49 @@
 #define MLPACK_METHODS_ANN_LOSS_FUNCTIONS_CROSS_ENTROPY_ERROR_IMPL_HPP
 
 // In case it hasn't yet been included.
-#include "cross_entropy_error.hpp"
+#include "binary_cross_entropy_loss.hpp"
 
 namespace mlpack {
 namespace ann /** Artificial Neural Network. */ {
 
 template<typename InputDataType, typename OutputDataType>
-CrossEntropyError<InputDataType, OutputDataType>::CrossEntropyError(
-    const double eps) : eps(eps)
+BCELoss<InputDataType, OutputDataType>::BCELoss(
+    const double eps, const bool reduction) : eps(eps), reduction(reduction)
 {
   // Nothing to do here.
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename PredictionType, typename TargetType>
 typename PredictionType::elem_type
-CrossEntropyError<InputDataType, OutputDataType>::Forward(
+BCELoss<InputDataType, OutputDataType>::Forward(
     const PredictionType& prediction,
     const TargetType& target)
 {
-  return -arma::accu(target % arma::log(prediction + eps) +
-      (1. - target) % arma::log(1. - prediction + eps));
+  typedef typename PredictionType::elem_type ElemType;
+
+  ElemType loss = -arma::accu(target % arma::log(prediction + eps) + 
+      (1. - target) % arma::log(1. - prediction + eps));;
+  if(reduction)
+    loss /= prediction.n_elem;
+  return loss;
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename PredictionType, typename TargetType, typename LossType>
-void CrossEntropyError<InputDataType, OutputDataType>::Backward(
+void BCELoss<InputDataType, OutputDataType>::Backward(
     const PredictionType& prediction,
     const TargetType& target,
     LossType& loss)
 {
   loss = (1. - target) / (1. - prediction + eps) - target / (prediction + eps);
+  if(reduction)
```

Comment:
```suggestion
  if (reduction)
```

Minor style fix insert space after `if`.